packages/launch/src/hooks/use-title.ts

```diff
@@ -2,33 +2,21 @@
  * External dependencies
  */
 import { useDispatch, useSelect } from '@wordpress/data';
-// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-// @ts-ignore
-import { useEntityProp } from '@wordpress/core-data';
+import { useState } from 'react';
 
 /**
  * Internal dependencies
  */
-import { LAUNCH_STORE } from '../stores';
+import { SITE_STORE } from '../stores';
 
-export function useTitle() {
-	const domain = useSelect( ( select ) => select( LAUNCH_STORE ).getSelectedDomain() );
-	const { setDomainSearch } = useDispatch( LAUNCH_STORE );
-	const [ title, setTitle ] = useEntityProp( 'root', 'site', 'title' );
-	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-	// @ts-ignore
-	const { saveEditedEntityRecord } = useDispatch( 'core' );
+export function useTitle( siteId: number ) {
+	const title = useSelect( ( select ) => select( SITE_STORE ).getSiteTitle( siteId ) );
+	const [ localStateTitle, setLocalStateTitle ] = useState( title );
```

Comment:
```suggestion
	const [ localStateTitle, setLocalStateTitle ] = useState( title || '' );
```
Making sure `title` is always a string here would prevent all the other checks for `undefined` (eg: [1](https://github.com/Automattic/wp-calypso/pull/46921/files#diff-2fb8ece2318e52dc186681406ed2610b204f21491260fe66331bbe668eabd625R57), [2](https://github.com/Automattic/wp-calypso/pull/46921/files#diff-8f2009196291a65edcfc3e93d032bc2aab593309be78da132e310207f0909025R46), [3](https://github.com/Automattic/wp-calypso/pull/46921/files#diff-cf261d1d8a4d4291e49ea46ec16d0a729c7dfde7be303f7fc590f6af00cbaa67R128)).